firefox_desktop/views/urlbar_events.view.lkml

```diff
@@ -1,223 +1,331 @@
+  dimension: num_history_impressions {
+    hidden:  yes
+    sql:  ARRAY_LENGTH(ARRAY(SELECT 1 FROM UNNEST(results) AS r WHERE r.product_result_type = "history"));;
+    type: number
+  }
+
+  dimension: num_navigational_impressions {
+    hidden:  yes
+    sql:  ARRAY_LENGTH(ARRAY(SELECT 1 FROM UNNEST(results) AS r WHERE r.product_result_type = "navigational"));;
+    type: number
+  }
+
+  dimension: num_open_tab_impressions {
+    hidden:  yes
+    sql:  ARRAY_LENGTH(ARRAY(SELECT 1 FROM UNNEST(results) AS r WHERE r.product_result_type = "open_tab"));;
+    type: number
+  }
+
+  dimension: num_pocket_collection_impressions {
+    hidden:  yes
+    sql:  ARRAY_LENGTH(ARRAY(SELECT 1 FROM UNNEST(results) AS r WHERE r.product_result_type = "pocket_collection"));;
+    type: number
+  }
+
+  dimension: num_quick_action_impressions {
+    hidden:  yes
+    sql:  ARRAY_LENGTH(ARRAY(SELECT 1 FROM UNNEST(results) AS r WHERE r.product_result_type = "quick_action"));;
+    type: number
+  }
+
+  dimension: num_search_engine_impressions {
+    hidden:  yes
+    sql:  ARRAY_LENGTH(ARRAY(SELECT 1 FROM UNNEST(results) AS r WHERE r.product_result_type = "search_engine"));;
+    type: number
+  }
+
+  dimension: num_suggest_addon_impressions {
+    hidden:  yes
+    sql:  ARRAY_LENGTH(ARRAY(SELECT 1 FROM UNNEST(results) AS r WHERE r.product_result_type = "suggest_add_on"));;
+    type: number
+  }
+
+  dimension: num_trending_suggestion_impressions {
+    hidden:  yes
+    sql:  ARRAY_LENGTH(ARRAY(SELECT 1 FROM UNNEST(results) AS r WHERE r.product_result_type = "trending_suggestion"));;
+    type: number
+  }
+
+  dimension: num_weather_impressions {
+    hidden:  yes
+    sql:  ARRAY_LENGTH(ARRAY(SELECT 1 FROM UNNEST(results) AS r WHERE r.product_result_type = "weather"));;
+    type: number
+  }
+
+  dimension: num_wikipedia_dynamic_impressions {
+    hidden:  yes
+    sql:  ARRAY_LENGTH(ARRAY(SELECT 1 FROM UNNEST(results) AS r WHERE r.product_result_type = "wikipedia_dynamic"));;
+    type: number
+  }
+
+  dimension: num_wikipedia_enhanced_impressions {
+    hidden:  yes
+    sql:  ARRAY_LENGTH(ARRAY(SELECT 1 FROM UNNEST(results) AS r WHERE r.product_result_type = "wikipedia_enhanced"));;
+    type: number
+  }
+
+  dimension: num_xchannels_addon_impressions {
+    hidden:  yes
+    sql:  ARRAY_LENGTH(ARRAY(SELECT 1 FROM UNNEST(results) AS r WHERE r.product_result_type = "xchannels_add_on"));;
+    type: number
+  }
 
   measure: event_count {
     hidden: yes
   }
 
-measure: urlbar_clicks {
-  group_label: "Urlbar Metrics"
-  description: "Count of clicks on any result shown in the urlbar dropdown menu"
-  sql: COUNTIF(${is_terminal} and ${session_action_type} = "engaged");;
-  type: number
-}
+  measure: urlbar_clicks {
+    group_label: "Urlbar Metrics"
+    description: "Count of clicks on any result shown in the urlbar dropdown menu"
+    sql: COUNT(DISTINCT(CASE WHEN ${is_terminal} and ${event_action} = "engaged" THEN ${event_id} ELSE NULL END));;
```

Comment:
Can we use COUNTIF like  in the previous SQL? I thought we still have 1 row per event.

Reply:
My intention was to future-proof this against:

- Someone requests a change to the auto-generated view in looker-hub so that it unpacks one of the ARRAY fields (experiments, results), and they didn't realize these metrics would be affected.
- You want to build off this view in looker-spoke-default to unpack an ARRAY field. Stakeholder sneakily accesses a hidden ARRAY field in Looker and unpacks it.

Reply:
Fair enough!